app/src/main/java/waiter/RouteDirector.java

```diff
@@ -0,0 +1,54 @@
+package waiter;
+
+public class RouteDirector {
+
+    public Route constructGetSimpleGet(RouteBuilder routeBuilder) {
+        return routeBuilder.newUp()
+                .url("/simple_get")
+                .method("GET")
+                .status("200 OK")
+                .headers("Content-Length:")
+                .body("")
+                .build();
+    }
+
+    public Route constructHeadSimpleGet(RouteBuilder routeBuilder) {
+        return routeBuilder.newUp()
+                .url("/simple_get")
+                .method("HEAD")
+                .status("200 OK")
+                .headers("Content-Length:")
+                .body("")
+                .build();
+    }
+    
+    public Route constructGetSimpleGetWithBody(RouteBuilder routeBuilder) {
+        return routeBuilder.newUp()
+                .url("/simple_get_with_body")
+                .method("GET")
+                .status("200 OK")
+                .headers("Content-Length:")
+                .body("Hello world")
+                .build();
+    }
+
+    public Route constructNoUrl(RouteBuilder routeBuilder) {
+        return routeBuilder.newUp()
+                .url("")
+                .method("")
+                .status("404 Not Found")
+                .headers("Content-Length:")
+                .body("")
+                .build();
+    }
+
+    public Route constructUrlExistsButNoMethodFound(RouteBuilder routeBuilder, String url) {
+        return routeBuilder.newUp()
+                .url(url)
+                .method("")
+                .status("404 Not Found")
+                .headers("Content-Length:")
```

Comment:
What happens if I don't specify this `Content-Length` header when creating a new route? Does this `Content-Length:` header need to be specified for every route? If so, how might we be able to remove this duplication?

Reply:
That is another really good point, as of right now if `Content-Length` is not specified we get that `no chunk, no close, no size. Assume close to signal end` section in the response that we were getting before. So I would say in order to get a proper response, it would want to be specified. As a result I was thinking of looking into making it a default configuration for any new route!

Reply:
I'm curious, if I have a route say `/foo` and it supports `GET`, `HEAD` and `POST`, does it mean I have to create 3 routes with each methods for the same resource?

Reply:
To Eli's point, we probably want to be able to set the content length dynamically. The current implementation suggests that `content length` and `body` can be set independent of each other which is not quite right imo.

Reply:
As of right now, you would have to create 3 routes with each methods for the same resource

We do set the content length dynamically, but that is true, it does suggest that they can be set independently, I will try to fix this so that the content-length header is not specified at all.

Reply:
+1 to the last point made by @mrfishball

I'll add that this feels to me like we're building the response when routes are being defined rather than when a request is received, parsed, etc.

Reply:
+1 to Eli's last point.

Reply:
Addressed in [this commit](https://github.com/jonmanahan/waiter/pull/18/commits/3bc8754c8fee547caf992d65eb1c11005cecf743)